command/agent/testagent.go

```diff
@@ -190,6 +191,9 @@ RETRY:
 	}
 	if failed {
 		a.Agent.Shutdown()
+		if i == 0 {
+			a.T.Fatalf("ran out of retries trying to start test agent")
+		}
```

Comment:
Before this could loop until the test timed out (10m by default) which is just cruel.

This whole TestAgent could use rewriting.